typing/typecore.ml

```diff
@@ -554,11 +554,11 @@ let rec build_as_type ~refine (env : Env.t ref) p =
           (fun () -> instance cty.ctyp_type)
       in
       (* This call to unify can't fail since the pattern is well typed. *)
```

Comment:
It is actually correct. The point of #10738 is that `~refine` is needed to make this unification succeed when the typing of the pattern succeed without generating an equation.

Reply:
My apologies. The comment is indeed stale. #10738 just changed from calling `unify` to `unify_pat_types`, so that we get a proper error message. `~refine` is irrelevant here.